Pinta.Tools/Tools/EraserTool.cs

```diff
@@ -32,98 +32,244 @@
 
 namespace Pinta.Tools
 {
-	public class EraserTool : BaseBrushTool
-	{		
-		private Point last_point = point_empty;
-
-		public EraserTool ()
-		{
-		}
-
-		protected override void OnBuildToolBar(Toolbar tb)
-		{
-			base.OnBuildToolBar(tb);
-
-			// Change the cursor when the BrushWidth is changed.
-			brush_width.ComboBox.Changed += (sender, e) => SetCursor (DefaultCursor);
-		}
-
-		#region Properties
-		public override string Name { get { return Catalog.GetString ("Eraser"); } }
-		public override string Icon { get { return "Tools.Eraser.png"; } }
-		public override string StatusBarText { get { return Catalog.GetString ("Left click to erase to transparent, right click to erase to secondary color. "); } }
-
-		public override Gdk.Cursor DefaultCursor {
-			get {
-				int iconOffsetX, iconOffsetY;
-				var icon = CreateIconWithShape ("Cursor.Eraser.png",
-				                                CursorShape.Ellipse, BrushWidth, 8, 22,
-				                                out iconOffsetX, out iconOffsetY);
+    public class EraserTool : BaseBrushTool
+    {       
+        protected enum EraserType
```

Comment:
Nothing derives from `EraserTool`, so I think the `protected` should be switched to `private`.